static/js/poll_modal.js

```diff
@@ -0,0 +1,76 @@
+import $ from "jquery";
+import {Sortable} from "sortablejs";
+
+import render_poll_modal_option from "../templates/poll_modal_option.hbs";
+
+export function update_option_delete_button_visibility($container, display_flag) {
+    const no_of_option_row = $container.find(".option-row").length;
+
+    // When there is only one option row, we hide its delete button
+    if (no_of_option_row === 1) {
+        // display_flag indicates whether another row will be added
+        // next, and in that case, we should show the delete button
+        if (display_flag === true) {
+            $container.find(".option-row .delete-option").show();
+        } else {
+            $container.find(".option-row .delete-option").hide();
+        }
+    }
+}
+
+function create_option_row(container) {
+    const row = render_poll_modal_option();
+    $(container).append(row);
+}
+
+function add_option_row(e) {
+    // if the option triggering the input event e is not the last,
+    // that is, its next sibling has the class `option-row`, we
+    // do not add a new option row and return from this function
+    // This handles a case when the next empty input row is already
+    // added and user is updating the above row(s).
+    if ($(e.target).parent().next().hasClass("option-row")) {
+        return;
+    }
+    const options_div = e.delegateTarget;
+    update_option_delete_button_visibility($(options_div), true);
+    create_option_row(options_div);
+}
+
+function delete_option_row(e) {
+    const $row = $(e.currentTarget).parent();
+    const $container = $row.parent();
+    $row.remove();
+    update_option_delete_button_visibility($container, false);
+}
+
+export function poll_options_setup() {
+    const $poll_options_list = $("#add-poll-form .poll-options-list .simplebar-content");
+
+    create_option_row($poll_options_list);
+    update_option_delete_button_visibility($poll_options_list, false);
+
+    Sortable.create($poll_options_list[0], {
+        onUpdate() {},
+        filter: "input",
+        preventOnFilter: false,
+    });
+
+    $poll_options_list.on("input", ".option-row input", add_option_row);
+    $poll_options_list.on("click", "button.delete-option", delete_option_row);
+}
+
+export function frame_poll_message_content() {
+    // if the question field is left empty we use the placeholder instead as
+    // the question is anyway editable by the poll creator in the widget
+    const question =
+        $("#poll-question-textarea").val().trim() ||
+        $("#poll-question-textarea").attr("placeholder");
+    const options = $(".poll-option-input")
+        .map(function () {
+            return $(this).val();
+        })
+        .toArray()
+        .filter(Boolean);
```

Comment:
Is the purpose of this filter to exclude the empty string? Do we need to trim these first?

Reply:
Oh, seems like I'd forgotten to trim this, partly since we do trim on the backend so the visible results aren't affected, but yes, the message we compose logically on the frontend did have empty spaces. Thanks for pointing this out, I'll trim the values here too!